src/core/ext/filters/client_channel/http_proxy_mapper.cc

```diff
@@ -77,12 +72,29 @@ bool ServerInCIDRRange(absl::string_view server_host,
   uint32_t mask_bits = 0;
   if (absl::SimpleAtoi(possible_cidr.second, &mask_bits)) {
     grpc_sockaddr_mask_bits(&*proxy_address, mask_bits);
-    return grpc_sockaddr_match_subnet(&*server_address, &*proxy_address,
+    return grpc_sockaddr_match_subnet(&server_address, &*proxy_address,
                                       mask_bits);
   }
   return false;
 }
 
+bool AddressIncluded(
+    const absl::StatusOr<grpc_resolved_address>& target_address,
```

Comment:
Suggest using `absl::optional<>` instead of `absl::StatusOr<>`.

Note that the caller on line 240 can use `StringToSockaddr(server_host, 0).value_or(absl::nullopt)`.

Reply:
Does not work, absl::nullopt_t can not be converted to grpc_resolved_address:
```
external/com_google_absl/absl/status/statusor.h:779:10: error: no viable conversion from returned value of type 'const absl::nullopt_t' to function return type 'grpc_resolved_address'
```

I think there's no harm in leaving StatusOr.

Reply:
I think using `StatusOr<>` seems like a very strange API for this function, since the function doesn't need the status; it just needs to know if there's an address that it should check for a CIDR range.

Same thing for `ServerInCIDRRange()` -- I think that function should go back to taking `const grpc_resolved_address&`, and the caller should not call it in the first place if there was no address to check.

Even if `.value_or()` doesn't work in this case, it's not hard for the caller to deal with this:

```
absl::StatusOr<grpc_resolved_address> address = StringToSockaddr(server_host, 0);
absl::optional<grpc_resolved_address> addr;
if (address.has_value()) addr = *address;
```

Reply:
Done